File: app/src/main/java/com/theappsolutions/boilerplate/util/other/RxUtils.java
package com.theappsolutions.boilerplate.util.other;

import io.reactivex.disposables.Disposable;

/**
 * @author Dmytro Yakovlev d.yakovlev@theappsolutions.com
 * @copyright (c) 2017 TheAppSolutions. (https://theappsolutions.com)
 */
public class RxUtils {

    public static void dispose(Disposable disposable) {
        if (disposable != null && !disposable.isDisposed()) {
            disposable.dispose();
        }
    }

}

File: app/src/main/java/com/theappsolutions/boilerplate/util/other/RxUtils.kt
package com.theappsolutions.boilerplate.util.other

import io.reactivex.disposables.Disposable

/**
 * @author Dmytro Yakovlev d.yakovlev@theappsolutions.com
 * @copyright (c) 2017 TheAppSolutions. (https://theappsolutions.com)
 */

fun dispose(disposable: Disposable?) {
    disposable?.let {
        if (!it.isDisposed) {
            it.dispose()
        }
    }
}

